thetis/solver.py

```diff
@@ -217,10 +217,8 @@ def compute_dt_2d(self, u_scale):
         l = inner(test, csize / u) * dx
         sp = {
             "snes_type": "ksponly",
-            "mat_type": "matfree",
-            "ksp_type": "preonly",
-            "pc_type": "python",
-            "pc_python_type": "firedrake.MassInvPC",
+            "ksp_type": "gmres",
+            "pc_type": "ilu",
```

Comment:
`ksp_type` should be `cg` (the operator is SPD), and you probably want `-pc_type bjacobi -sub_pc_type ilu` to work right in parallel.